core/src/test/java/org/elasticsearch/index/seqno/LocalCheckpointTrackerTests.java

```diff
@@ -95,6 +98,14 @@ public void testSimpleReplica() {
         assertThat(tracker.getCheckpoint(), equalTo(2L));
     }
 
+    public void testLazyInitialization() {
+        /*
+         * Previously this would allocate the entire chain of bit sets to the one for the sequence number being marked; for very large
+         * sequence numbers this could lead to excessive memory usage resulting in out of memory errors.
+         */
+        tracker.markSeqNoAsCompleted(randomNonNegativeLong());
```

Comment:
lol. can we assert we allocated just one array?

Reply:
I pushed b2cbfc55bb2e319c74215560aa5986d1d023612a.